### PR TITLE
Update detection bundle with TensorRT

### LINKED ISSUE #385: Support the lung_nodule_ct_segmentation bundle with TensorRT
**Is your feature request related to a problem? Please describe.**
The lung_nodule_ct_segmentation bundle doesn't support the TensorRT inference for now, since there are some pythonic logics in this bundle. It would be better if the bundle could be accelerated by the TensorRT conversion. The blocking issues are:
1. The `detector` takes a torch `network` as input, which has some necessary attributes like `num_anchors`, `num_class`, `spatial_dim` and so on. But the model exported by TensorRT doesn't retain these attributes. 
2. The `detector` have two ways to do inference. If the input size is larger than the presetting `infer_patch_size`, a `slide_window_inferer` is chosen for the inference. Otherwise, the input is directly fed to the network for inference. This causes the inputs of the network have different spatial sizes,   `infer_patch_size` or the input size (if the input size is smaller than the  `infer_patch_size`). However the TensorRT model only takes fixed size inputs (except for the batch size).

**Describe the solution you'd like**
1. Update the `detector` with input parameters to replace the `network` attributes.
2. Add a parameter to contoll whether force the inference with `slide_window_inferer`.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by binliunls**
Fixes #385  .

### Description
Update the detection bundle with ONNX-TensorRT conversion.

### Status
**Work in progress**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: models/lung_nodule_ct_detection/configs/metadata.json
Comment (yiheng-wang-nv) [inline models/lung_nodule_ct_detection/configs/metadata.json:5]: could you add  this bundle into:
https://github.com/Project-MONAI/model-zoo/blob/dev/ci/bundle_custom_data.py#L39

File: models/lung_nodule_ct_detection/scripts/detection_inferer.py
Comment (yiheng-wang-nv) [inline models/lung_nodule_ct_detection/scripts/detection_inferer.py]: Hi @Can-Zhao , may need your help to review this file. I'm not sure if the naming here is suitable. Thanks in advance.

File: models/lung_nodule_ct_detection/docs/README.md
Comment (yiheng-wang-nv) [inline models/lung_nodule_ct_detection/docs/README.md]: Otherwise, if given an input ...

File: models/lung_nodule_ct_detection/scripts/detection_inferer.py
Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/scripts/detection_inferer.py]: Looks good to me! Thank you!

**Comment by Can-Zhao** [comment]
@binliunls Do we need to retrain in this case? Or would the existing trained weights work with TensorRT inference?

**Review by Can-Zhao — APPROVED** [review]

File: models/lung_nodule_ct_detection/scripts/detection_inferer.py
Comment (Nic-Ma) [reply models/lung_nodule_ct_detection/scripts/detection_inferer.py]: I feel this arg name is very confusing, would suggest to others, for example: `force_sliding_window`.
@binliunls @Can-Zhao What do you think?

Thanks.

File: models/lung_nodule_ct_detection/configs/inference.json
Comment (Nic-Ma) [inline models/lung_nodule_ct_detection/configs/inference.json]: May I know why you put these 3 parameters as global vars here?

Thanks.

File: models/lung_nodule_ct_detection/scripts/detection_inferer.py
Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/scripts/detection_inferer.py]: Yes, force_sliding_window sounds better

File: models/lung_nodule_ct_detection/configs/inference.json
Comment (binliunls) [reply models/lung_nodule_ct_detection/configs/inference.json]: Hi @Nic-Ma ,
     Because the `spatial_dims` and `num_classes` parameters are used in both `network_def` and `detector`, I think it would be better to use a parameter to control them instead of directly specifying the value in the `network_def` and `detector`.
     For the `force_inferer` (or `force_sliding_window`) parameter, it is set as a global vars so that users can easily use it in the command line, like `python -m monai.bundle run ... --force_sliding_window "True"`

Thanks,
Bin

**Comment by binliunls** [comment]
> @binliunls Do we need to retrain in this case? Or would the existing trained weights work with TensorRT inference?

Hi Can, no need to retrain the model, the TensorRT model only exports the trained model to a more efficient one and does the inference with it.

File: models/lung_nodule_ct_detection/docs/README.md
Comment (yiheng-wang-nv) [inline models/lung_nodule_ct_detection/docs/README.md:66]: Hi @zephyrie , could you please help to review this section? Thanks in advance!

**Comment by binliunls** [comment]
Hi @Can-Zhao ,
       Could you please take a look at [this line](https://github.com/Project-MONAI/model-zoo/blob/ef6f861b2c14add848ba45d55dd1bc7c8a31ea78/models/lung_nodule_ct_detection/scripts/detection_inferer.py#L32). The `*args, **kwargs` part seems useless. I would like to update it from `def __init__(self, detector: RetinaNetDetector, *args, **kwargs) -> None` to `def __init__(self, detector: RetinaNetDetector, force_sliding_window: bool=False) -> None`. What do you think about it.

Thanks,
Bin

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao , Could you please take a look at [this line](https://github.com/Project-MONAI/model-zoo/blob/ef6f861b2c14add848ba45d55dd1bc7c8a31ea78/models/lung_nodule_ct_detection/scripts/detection_inferer.py#L32). The `*args, **kwargs` part seems useless. I would like to update it from `def __init__(self, detector: RetinaNetDetector, *args, **kwargs) -> None` to `def __init__(self, detector: RetinaNetDetector, force_sliding_window: bool=False) -> None`. What do you think about it.
> 
> Thanks, Bin

Yes, it is useless.

**Review by yiheng-wang-nv — APPROVED** [review]
LGTM, after @zephyrie 's double confirm we can merge this PR

**Comment by yiheng-wang-nv** [comment]
Update: @zephyrie reviewed the readme, and sent me some suggestions offline, I will push a new commit to this PR to address that.
For other bundles that have similar descriptions, I will modify them as well within this PR.

File: models/pathology_tumor_detection/configs/metadata.json
Comment (binliunls) [inline models/pathology_tumor_detection/configs/metadata.json:5]: Should the monai version of this bundle be changed to `1.2.0rc5`? Same question for other bundles that don't change to rc5.
Thanks,
Bin

**Comment by binliunls** [comment]
LGTM, thanks for the update.

**Pull request merged**
